### PR TITLE
Pass redirect through to client

### PR DESCRIPTION
**Pull request by ASankaran**
Addresses #45

Instead of following redirects pass them through to the client, so that the client can deal with the redirect as they want.

- [x] Make sure to tag the issue you are addressing in your comments if applicable. 
- [x] Make sure any tests pass
- [x] Make sure you stick to rough style guidelines
- [ ] Have you added the license header to any new files? 
- [ ] Tag a reviewer 
- [ ] Select target milestone
- [ ] Make sure to document any new features or API changes

### CODE REVIEW

**Comment by narendasan** [comment]
Yeah I think so as well @ASankaran can you implement those and also add a debug message for when the resource is found so it’s easier to track the flow of the request?

**Comment by narendasan** [comment]
Also @bcongdon worth supporting delete too?

**Comment by ASankaran** [comment]
Added POST and PUT support and the logging. LMK if you want support on DELETE as well.

File: proxy/PUT.go [hunk 1]
```diff
@@ -173,15 +184,31 @@ func xmlPUT(r *http.Request, w http.ResponseWriter, url string, token string, da
...
+		logger.Log(logger.ERR, err.Error())
+		return
+	} else if resp.StatusCode == http.StatusFound {
+		logger.Log(logger.DEBUG, "SERVICE RETURNED REDIRECT")
+		w.Header().Set("Location", resp.Header.Get("Location"))
+		w.WriteHeader(http.StatusFound)
+		return
+	} else if resp.StatusCode != http.StatusOK {
+		logger.Log(logger.WARN, "SERVICE FAILED - SERVICE RETURNED STATUS "+http.StatusText(resp.StatusCode))
+		w.Header().Set("Content-Type", "<?xml version=\"1.0\" encoding=\"UTF-8\"?>")
```
Comment (bcongdon) [inline proxy/PUT.go]: Is this supposed to be hard-coded?

File: proxy/PUT.go [hunk 2]
```diff
@@ -173,15 +184,31 @@ func xmlPUT(r *http.Request, w http.ResponseWriter, url string, token string, da
...
+		},
+	}
 	resp, err := client.Do(req)
 	logger.LogResp(logger.DEBUG, resp)
 
-	if err != nil || resp.StatusCode != http.StatusOK {
-		invalidPUT(w, err)
-		logger.Log(logger.ERR, fmt.Sprintf("Failed request:%v", err))
+	if err != nil {
+		//TODO: INVALID PUT HANDLER
```
Comment (bcongdon) [inline proxy/PUT.go]: Is this TODO supposed to be left in here?

File: proxy/PUT.go (hunk 1 above)
Comment (bcongdon) [inline proxy/PUT.go]: Is this supposed to be hard-coded?

**Comment by bcongdon** [comment]
LGTM, I'd like @narendasan to sign off before merging though

**Pull request merged**